File: backend/app/schemas/lens.py
from typing import List

from pydantic import BaseModel

from app.schemas.movie import MovieOut


class LensVisualLabelOut(BaseModel):
    key: str
    label: str
    score: float
    genres: List[str]
    mood: str


class LensRecommendationOut(BaseModel):
    movie: MovieOut
    score: float
    reason: str
    visual_similarity: float
    matched_genres: List[str]


class LensAnalyzeOut(BaseModel):
    mode: str
    title: str
    description: str
    visual_labels: List[LensVisualLabelOut]
    matched_genres: List[str]
    recommendations: List[LensRecommendationOut]

File: android/app/src/main/java/com/app/movieit/data/model/Lens.kt
package com.app.movieit.data.model

import com.google.gson.annotations.SerializedName

data class LensVisualLabel(
    val key: String,
    val label: String,
    val score: Float,
    val genres: List<String> = emptyList(),
    val mood: String?
)

data class LensRecommendation(
    val movie: Movie,
    val score: Float,
    val reason: String,
    @SerializedName("visual_similarity") val visualSimilarity: Float,
    @SerializedName("matched_genres") val matchedGenres: List<String> = emptyList()
)

data class LensAnalyzeResponse(
    val mode: String,
    val title: String,
    val description: String,
    @SerializedName("visual_labels") val visualLabels: List<LensVisualLabel> = emptyList(),
    @SerializedName("matched_genres") val matchedGenres: List<String> = emptyList(),
    val recommendations: List<LensRecommendation> = emptyList()
)
